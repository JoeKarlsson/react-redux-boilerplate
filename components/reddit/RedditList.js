'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import RedditItem from './RedditItem';

class RedditList extends React.Component {
  render() {
    var redditListNode = this.props.redditData.map(function(redditDataItem, index){
      return (
        <RedditItem
          id={redditDataItem.data.id}
          index={index}
          author={redditDataItem.data.author}
          key={redditDataItem.data.id} >
          {redditDataItem.data.title}
        </RedditItem>
      )
    })
    return (
      <div className='redditList'>
        <h2>Reddit List</h2>
        { redditListNode }
      </div>
    )
  }
};

export default RedditList;
