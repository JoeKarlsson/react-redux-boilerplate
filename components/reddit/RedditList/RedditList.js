import React from 'react';
import PropTypes from 'prop-types';
import RedditItem from '../RedditItem/RedditItem';
import styles from './RedditList.scss';

class RedditList extends React.Component {
  render() {
    var redditListNode = this.props.redditData.map(function(redditDataItem, index){
      return (
        <RedditItem
          id={redditDataItem.data.id}
          index={index}
          author={redditDataItem.data.author}
          key={redditDataItem.data.id} >
          {redditDataItem.data.title}
        </RedditItem>
      )
    })
    return (
      <div className={styles.RedditList}>
        <h2>Reddit List</h2>
        { redditListNode }
      </div>
    )
  }
};

export default RedditList;
