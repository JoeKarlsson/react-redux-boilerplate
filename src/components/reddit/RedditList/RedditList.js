import React from 'react';
import PropTypes from 'prop-types';
import RedditItem from '../RedditItem/RedditItem';
import styles from './RedditList.scss';

class RedditList extends React.Component {

	static redditItemNodeBuilder(redditDataItem, index) {
		return (
			<RedditItem
				id={redditDataItem.data.id}
				index={index}
				author={redditDataItem.data.author}
				key={redditDataItem.data.id}
			>
				{redditDataItem.data.title}
			</RedditItem>
		);
	}

	render() {
		const redditListNode = this.props.redditData.map(RedditList.redditItemNodeBuilder);

		return (
			<div className={styles.RedditList}>
				<h2>Reddit List</h2>
				{ redditListNode }
			</div>
		);
	}
}

RedditList.propTypes = {
	redditData: PropTypes.arrayOf(PropTypes.object).isRequired,
};

RedditList.defaultProps = {
	redditData: [],
};

export default RedditList;
