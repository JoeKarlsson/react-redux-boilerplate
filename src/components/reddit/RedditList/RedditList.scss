$green: green;

.RedditList {
	background-color: $green;
}
