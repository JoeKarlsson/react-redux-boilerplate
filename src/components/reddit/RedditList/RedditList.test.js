import React from 'react';
import renderer from 'react-test-renderer';
import {
	MemoryRouter,
} from 'react-router-dom';
import RedditList from './RedditList';

test('RedditList page snapshot matches', () => {
	const component = renderer.create(
		<MemoryRouter><RedditList /></MemoryRouter>,
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
