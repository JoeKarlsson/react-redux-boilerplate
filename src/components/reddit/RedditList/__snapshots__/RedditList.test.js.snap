// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedditList page snapshot matches 1`] = `
<div
  className="RedditList"
>
  <h2>
    Reddit List
  </h2>
</div>
`;
